### PR TITLE
Download reviews from modal

### PR DESCRIPTION
**Pull request by MrRob100**
https://github.com/pkp/pkp-lib/issues/9873

### CODE REVIEW

**Comment by MrRob100** [comment]
@jardakotesovec, for the download reviews feature I have had to add javascript to https://github.com/MrRob100/pkp-lib/blob/download-reviews-from-modal/templates/controllers/grid/users/reviewer/readReview.tpl . Is there a better way of including it in a js file rather than embedded? The only ways I can see are for forms. I would do it in vue but our select component wont match Devikas design shown here https://github.com/pkp/pkp-lib/issues/9873

**Comment by jardakotesovec** [comment]
@MrRob100 
Apologies for delay.

It would be awesome if you could use this Vue.js component - https://main--6555d3db80418bb1681b8b17.chromatic.com/?path=/docs/components-dropdownactions--docs

Also if you could make sure that the indentation in readReview.tpl does not change so its lot easier to see whats new/different there.

And while you are on it if you could rebase, so it works correctly with latest main branch on ojs and ui-library.

Thank you!

**Comment by jardakotesovec** [comment]
@Vitaliy-1 Could you have a look on the backend part of this. There are are some new dependencies to generate the pdf etc. And I think you have some experience with that?

File: classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php [hunk 1]
```diff
@@ -962,6 +967,371 @@ public function sendEmail($args, $request)
...
+     * @param array $args
+     * @param PKPRequest $request
+     *
+     * @return void
+     */
+    function exportPDF($args, $request)
+    {
+        $submissionCommentDao = DAORegistry::getDAO('SubmissionCommentDAO'); /* @var $submissionCommentDao SubmissionCommentDAO */
+        $context = $request->getContext();
+        $submissionId = $request->getUserVar('submissionId');
```
[the remark this answers is not in the document]
Comment (Vitaliy-1) [reply classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php:982]: The general idea is that we check if user is permitted to access the submission or other entity.

**Review by Vitaliy-1 — CHANGES_REQUESTED** [review]
Hi @MrRob100. I left a couple of comments. My main suggestion is to add UFT-8 support for generated PDFs, is it something doable? Thanks, it looks great!

File: classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrRob100) [reply classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php:979]: Hi @Vitaliy-1 Changes are in but `$reviewAssignment = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_REVIEW_ASSIGNMENT);` seems to be null for any user. Could you have a look at this please?

File: classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php [hunk 2]
```diff
@@ -962,6 +967,371 @@ public function sendEmail($args, $request)
...
+     * Export review as XML
+     *
+     * @param array $args
+     * @param PKPRequest $request
+     *
+     * @return void
+     */
+    public function exportXML($args, $request) {
+        $submissionId = $request->getUserVar('submissionId');
+        $reviewId = $request->getUserVar('reviewAssignmentId');
```
[the remark this answers is not in the document]
Comment (MrRob100) [reply classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.php]: @Vitaliy-1 However the logic has been moved to a controller now so works differently

File: api/v1/reviews/PKPReviewController.php [hunk 3]
```diff
@@ -225,4 +257,402 @@ public function getHistory(Request $illuminateRequest): JsonResponse
...
+        $context = $request->getContext();
+        $submissionId = $request->getUserVar('submissionId');
+        $reviewId = $request->getUserVar('reviewAssignmentId');
+        $reviewAssignment = Repo::reviewAssignment()->get($reviewId);
+        $submissionComments = $submissionCommentDao->getReviewerCommentsByReviewerId($submissionId, $reviewAssignment->getReviewerId(), $reviewId, true);
+        $submissionCommentsPrivate = $submissionCommentDao->getReviewerCommentsByReviewerId($submissionId, $reviewAssignment->getReviewerId(), $reviewId, false);
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+        $title = $submission->getCurrentPublication()->getLocalizedTitle(null, 'html');
+        $cleanTitle = str_replace("&nbsp;", " ", strip_tags($title));
+        $mpdf = new Mpdf([
```
Comment (ewhanson) [inline api/v1/reviews/PKPReviewController.php:277]: To aid in readability of this and the XML method below, could you break out the actual PDF/XML generation into separate functions so the API handling vs the PDF/XML generation are a bit easier to follow and separate? If these endpoints/controllers grow in the future, I can see wanting to extract the document generation out to separate classes as well.

**Review by ewhanson — CHANGES_REQUESTED** [review]

**Review by ewhanson — CHANGES_REQUESTED** [review]

**Pull request merged**